File: C++/BruteForce/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int supo1[5] = { 1, 2, 3, 4, 5 };
int supo2[8] = { 2, 1, 2, 3, 2, 4, 2, 5 };
int supo3[10] = { 3, 3, 1, 1, 2, 2, 4, 4, 5, 5 };
int cnt[3];

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int max_cnt = 0;

    for (int i = 0; i < answers.size(); i++) {
        if (supo1[i % 5] == answers[i]) cnt[0]++;
        if (supo2[i % 8] == answers[i]) cnt[1]++;
        if (supo3[i % 10] == answers[i]) cnt[2]++;
    }
    max_cnt = max(cnt[0], max(cnt[1], cnt[2]));

    for (int i = 0; i < 3; i++) {
        if (max_cnt == cnt[i]) answer.push_back(i + 1);
    }

    return answer;
}

File: Python/BruteForce/모의고사.py
def solution(answers):
    cnt = [0, 0, 0]
    answer = []

    supo1 = [1, 2, 3, 4, 5]
    supo2 = [2, 1, 2, 3, 2, 4, 2, 5]
    supo3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    for idx in range(len(answers)):
        if answers[idx] == supo1[idx%5]:
            cnt[0] = cnt[0] + 1
        if answers[idx] == supo2[idx%8]:
            cnt[1] = cnt[1] + 1
        if answers[idx] == supo3[idx%10]:
            cnt[2] = cnt[2] + 1
    
    for i in range(len(cnt)):
        if cnt[i] == max(cnt):
            answer.append(i + 1)

    return answer

answers1 = [1, 2, 3, 4, 5]
answers2 = [1, 3, 2, 4, 2]
print(solution(answers2))